addons/mrp/models/mrp_workorder.py

```diff
@@ -345,7 +345,7 @@ def _float_duration_to_second(duration):
             delta_duration = new_order_duration - old_order_duration
 
             if delta_duration > 0:
-                if order.state not in ('progress', 'done'):
+                if order.state not in ('progress', 'done', 'cancel'):
```

Comment:
I find it weird that a cancelled workorder would have a `mrp.workcenter.productivity` with a full `duration` based on its expected duration. :confused: